src/main/java/com/aws/greengrass/deployment/DeploymentConfigMerger.java

```diff
@@ -313,6 +324,24 @@ public void reinstallBrokenServices() throws ServiceLoadException {
             }
         }
 
+        /**
+         * Close all unloadable service instances, initialize each service with updated config and add to context.
+         *
+         * @throws ServiceLoadException when any service to be replaced could not be located
+         */
+        public void replaceUnloadableService() throws ServiceLoadException {
+            for (String serviceName : alreadyUnloadableServices) {
+                try {
+                    kernel.locate(serviceName).close().get(30, TimeUnit.SECONDS);
```

Comment:
When calling close() here, it waits for depernder services to exit. How is the depender services notified that it has to exit?